src/control/security/config_test.go

```diff
@@ -0,0 +1,209 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package security
+
+import (
+	"os"
+	"strings"
+	"bytes"
+	"testing"
+)
+
+func InsecureTC() *TransportConfig{
+	return &TransportConfig{
+		AllowInsecure: true,
+	}
+}
+func BadTC() *TransportConfig{
+	return &TransportConfig{
+		AllowInsecure: false,
+		CertificateConfig: CertificateConfig{
+			CARootPath: "testdata/certs/daosCA.crt",
+			CertificatePath: "testdata/certs/bad.crt",
+			PrivateKeyPath: "testdata/certs/bad.key",
+			tlsKeypair:      nil,
+			caPool:          nil,
+		},
+	}
+}
+func ServerTC() *TransportConfig {
+	return &TransportConfig{
+		AllowInsecure: false,
+		CertificateConfig: CertificateConfig{
+			CARootPath: "testdata/certs/daosCA.crt",
+			CertificatePath: "testdata/certs/server.crt",
+			PrivateKeyPath: "testdata/certs/server.key",
+			tlsKeypair:      nil,
+			caPool:          nil,
+		},
+	}
+}
+func AgentTC() *TransportConfig {
+	return &TransportConfig{
+		AllowInsecure: false,
+		CertificateConfig: CertificateConfig{
+			CARootPath: "testdata/certs/daosCA.crt",
+			CertificatePath: "testdata/certs/agent.crt",
+			PrivateKeyPath: "testdata/certs/agent.key",
+			tlsKeypair:      nil,
+			caPool:          nil,
+		},
+	}
+}
+func SetupTCFilePerms(t *testing.T, conf *TransportConfig) {
+	if err := os.Chmod(conf.CARootPath, SafeCertPerm); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.Chmod(conf.CertificatePath, SafeCertPerm); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.Chmod(conf.PrivateKeyPath, SafeKeyPerm); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func ValidateInsecure (t *testing.T, c *TransportConfig, err error){
+	if err != nil {
+		t.Fatal(err)
+	}
+	if c.tlsKeypair != nil || c.caPool != nil {
+		t.Fatal("insecure config loaded certs")
+	}
+}
+func ValidateGood (t *testing.T, c *TransportConfig, err error){
+	if err != nil {
+		t.Fatal(err)
+	}
+	if c.tlsKeypair == nil || c.caPool == nil {
+		t.Fatal("certs did not load yet returned no error")
+	}
+}
+func ValidateBad (t *testing.T, c *TransportConfig, err error){
+	if err == nil {
+		t.Fatal(err)
+	}
+}
+func ValidateNil (t *testing.T, c *TransportConfig, err error){
+	if err != nil &&
+		strings.Compare(err.Error(), "nil TransportConfig") != 0 {
+		t.Fatal(err)
+	}
+}
+func TestPreLoadCertData(t *testing.T) {
+	insecureTC := InsecureTC()
+	serverTC := ServerTC()
+	badTC := BadTC()
+
+	// Setup permissions for tests below.
+	SetupTCFilePerms(t, serverTC)
+	SetupTCFilePerms(t, badTC)
+
+	testCases := []struct {
```

Comment:
For future reference, you can also organize these as a map[string]struct{ ... }. The benefit to this is that the testcase name becomes the map key instead of a struct member, and it also makes the test runs less-deterministic (which can sometimes be a good thing for smoking out weirdness).